src/warzone/src/warzone/model/GameContext.java

```diff
@@ -67,6 +72,19 @@ public static GameContext getGameContext() {
 		return GAME_CONTEXT;
 	}
 	
+	/**
+	 * This method can return the logEntryBuffer instance and create a new one if
+	 * it is null.
+	 * @return the logEntryBuffer instance
+	 */
+	public static LogEntryBuffer getLogEntryBuffer() {
+		if(LOG_ENTRY_BUFFER == null) {
+			LOG_ENTRY_BUFFER= new LogEntryBuffer();
+			LOG_ENTRY_BUFFER.attach(new LogService());
+		}
+		return LOG_ENTRY_BUFFER;
+	}
+	
```

Comment:
move the singleton to LogEntryBuffer class, and only add datamember in the GameContext